### PR TITLE
refactor: rename CreateResource to CreateOrUpdateResource

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

This PR refactors the async interface `CreateResource` method to `CreateOrUpdateResource` to more accurately reflect what's actually going on.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: .codespellignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ witht
 geting
 ot
 intepreted
+updat
```
Comment (mboersma) [inline .codespellignore:6]: But the linter is ok with `creat`? Lol.

File: azure/services/async/async.go [hunk 2]
```diff
@@ -132,20 +133,25 @@ func (s *Service) CreateResource(ctx context.Context, spec azure.ResourceSpecGet
 	}
 
 	// Create or update the resource with the desired parameters.
-	log.V(2).Info("creating resource", "service", serviceName, "resource", resourceName, "resourceGroup", rgName)
+	logMessageVerbPrefix := "creat"
```
Comment (mboersma) [inline azure/services/async/async.go:136]: I love the more specific log messages here. (Good thing we don't have to i18n this.)

**Review by mboersma — COMMENTED** [review]
/lgtm

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**